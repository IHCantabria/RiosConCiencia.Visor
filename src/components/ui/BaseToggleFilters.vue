<template>
  <div class="toggle-filters">
    <vs-dropdown
      title="Advanced Filters"
      class="toggler"
      vs-trigger-click
      vs-custom-content
    >
      <div class="toggle-icon">
        <object
          :data="require('@/assets/advanced.svg')"
          type="image/svg+xml"
          style="pointer-events: none;"
        />
      </div>
      <vs-dropdown-menu>
        <div class="switch-container">
          <label class="switch">Only Spatial Filters</label>
          <vs-switch
            v-model="onlySpatialFiltersToggle"
            vs-icon="done"
            vs-icon-off="block"
            class="switch"
          >
          </vs-switch>
        </div>
        <div class="switch-container">
          <label class="switch">Search Products</label>
          <vs-switch
            v-model="searchProductsToggle"
            vs-icon="done"
            vs-icon-off="block"
            class="switch"
          >
          </vs-switch>
        </div>
        <div class="switch-container">
          <label class="switch">Search Buoys</label>
          <vs-switch
            v-model="searchBuoysToggle"
            vs-icon="done"
            vs-icon-off="block"
            class="switch"
          >
          </vs-switch>
        </div>
        <!--Commented to hide station logic
        <div class="switch-container">
          <label class="switch">Search Stations</label>
          <vs-switch
            v-model="searchStationsToggle"
            vs-icon="done"
            vs-icon-off="block"
            class="switch"
          >
          </vs-switch>
        </div> -->
      </vs-dropdown-menu>
    </vs-dropdown>
  </div>
</template>
<script>
export default {
  data() {
    return {
      onlySpatialFiltersToggle: false,
      searchProductsToggle: true,
      searchBuoysToggle: true,
      //SearchStationsToggle is False to hide station logic
      searchStationsToggle: false
    };
  },
  computed: {
    areOnlySpatialFiltersActive() {
      return this.onlySpatialFiltersToggle;
    },
    isSearchProductsActive() {
      return this.searchProductsToggle;
    },
    isSearchBuoysActive() {
      return this.searchBuoysToggle;
    },
    isSearchStationsActive() {
      return this.searchStationsToggle;
    }
  },
  watch: {
    areOnlySpatialFiltersActive(newValue) {
      this.$emit("only-spatial-filters-actived", newValue);
    },
    isSearchProductsActive(newValue) {
      this.$emit("products-actived", newValue);
    },
    isSearchBuoysActive(newValue) {
      this.$emit("buoys-actived", newValue);
    },
    isSearchStationsActive(newValue) {
      this.$emit("stations-actived", newValue);
    }
  }
};
</script>
<style lang="scss" scoped>
.toggler {
  &:hover {
    cursor: pointer;
  }
}
.toggle-icon {
  width: 26px;
  padding: 5px;
  margin: 0 0 0 4px;
  &:hover {
    cursor: pointer;
  }
}
.switch {
  margin: 2px;
}
.switch-container {
  justify-content: space-between;
  display: flex;
  flex-flow: nowrap row;
}
</style>
