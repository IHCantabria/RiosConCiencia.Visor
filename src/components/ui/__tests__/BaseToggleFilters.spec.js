import { shallowMount, createLocalVue } from "@vue/test-utils";
import BaseToggleFilters from "../BaseToggleFilters.vue";
import Vuesax from "vuesax";
import merge from "lodash.merge";

const localVue = createLocalVue();
localVue.use(Vuesax);

function createWrapper(overrides) {
  const defaultMountingOptions = {
    localVue
  };
  return shallowMount(
    BaseToggleFilters,
    merge(defaultMountingOptions, overrides)
  );
}

describe("baseToggleFilters.vue", () => {
  //UNIT TESTS
  it("computed areBaseFiltersActive updated when baseFiltersToggle change", () => {
    expect.assertions(2);
    const wrapper = createWrapper();
    expect(wrapper.vm.areOnlySpatialFiltersActive).toBe(false);
    wrapper.setData({ onlySpatialFiltersToggle: true });
    expect(wrapper.vm.areOnlySpatialFiltersActive).toBe(true);
  });
  it("filters-actived is emitted when watcher areBaseFiltersActive was actived", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    wrapper.setData({ onlySpatialFiltersToggle: true });
    expect(wrapper.emitted()["only-spatial-filters-actived"]).toStrictEqual([
      [true]
    ]);
  });
  it("computed isSearchProductsActive updated when searchProductsToggle change", () => {
    expect.assertions(2);
    const wrapper = createWrapper();
    expect(wrapper.vm.isSearchProductsActive).toBe(true);
    wrapper.setData({ searchProductsToggle: false });
    expect(wrapper.vm.isSearchProductsActive).toBe(false);
  });
  it("products-actived is emitted when watcher isSearchProductsActive was actived", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    wrapper.setData({ searchProductsToggle: false });
    expect(wrapper.emitted()["products-actived"]).toStrictEqual([[false]]);
  });
  it("computed isSearchBuoysActive updated when searchBuoysToggle change", () => {
    expect.assertions(2);
    const wrapper = createWrapper();
    expect(wrapper.vm.isSearchBuoysActive).toBe(true);
    wrapper.setData({ searchBuoysToggle: false });
    expect(wrapper.vm.isSearchBuoysActive).toBe(false);
  });
  it("buoys-actived is emitted when watcher isSearchBuoysActive was actived", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    wrapper.setData({ searchBuoysToggle: false });
    expect(wrapper.emitted()["buoys-actived"]).toStrictEqual([[false]]);
  });
  it("computed isSearchStationsActive updated when searchStationsToggle change", () => {
    expect.assertions(2);
    const wrapper = createWrapper();
    expect(wrapper.vm.isSearchStationsActive).toBe(true);
    wrapper.setData({ searchStationsToggle: false });
    expect(wrapper.vm.isSearchStationsActive).toBe(false);
  });
  it("fstations-actived is emitted when watcher isSearchStationsActive was actived", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    wrapper.setData({ searchStationsToggle: false });
    expect(wrapper.emitted()["stations-actived"]).toStrictEqual([[false]]);
  });

  //SNAPSHOT TESTS
  it("renders correctly", () => {
    expect.assertions(1);
    const wrapper = createWrapper({});
    expect(wrapper.element).toMatchSnapshot();
  });
});
