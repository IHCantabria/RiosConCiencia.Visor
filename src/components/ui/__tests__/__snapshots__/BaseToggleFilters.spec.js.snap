// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseToggleFilters.vue renders correctly 1`] = `
<div
  class="toggle-filters"
>
  <vs-dropdown-stub
    class="toggler"
    color="primary"
    title="Advanced Filters"
    vscustomcontent="true"
    vstriggerclick="true"
  >
    <div
      class="toggle-icon"
    >
      <object
        data=""
        style="pointer-events: none;"
        type="image/svg+xml"
      />
    </div>
     
    <vs-dropdown-menu-stub>
      <div
        class="switch-container"
      >
        <label
          class="switch"
        >
          Only Spatial Filters
        </label>
         
        <vs-switch-stub
          class="switch"
          color="primary"
          iconpack="material-icons"
          vsicon="done"
          vsiconoff="block"
        />
      </div>
       
      <div
        class="switch-container"
      >
        <label
          class="switch"
        >
          Search Products
        </label>
         
        <vs-switch-stub
          class="switch"
          color="primary"
          iconpack="material-icons"
          value="true"
          vsicon="done"
          vsiconoff="block"
        />
      </div>
       
      <div
        class="switch-container"
      >
        <label
          class="switch"
        >
          Search Buoys
        </label>
         
        <vs-switch-stub
          class="switch"
          color="primary"
          iconpack="material-icons"
          value="true"
          vsicon="done"
          vsiconoff="block"
        />
      </div>
       
      <div
        class="switch-container"
      >
        <label
          class="switch"
        >
          Search Stations
        </label>
         
        <vs-switch-stub
          class="switch"
          color="primary"
          iconpack="material-icons"
          value="true"
          vsicon="done"
          vsiconoff="block"
        />
      </div>
    </vs-dropdown-menu-stub>
  </vs-dropdown-stub>
</div>
`;
